### PR TITLE
Migrate API types and configs

### PR DESCRIPTION
**Pull request by sjberman**
Moving any API or config types that are used between internal repositories to Github to remove the internal dependencies. Now all components can pull purely from Github.

- [x] I have read the [CONTRIBUTING](https://github.com/nginxinc/nginx-service-mesh/blob/main/CONTRIBUTING.md) doc
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] I have checked that all unit tests pass after adding my changes
- [ ] I have updated necessary documentation
- [x] I have rebased my branch onto main
- [x] I will ensure my PR is targeting the main branch and pulling from my branch from my own fork

### CODE REVIEW

**Review by chase-kiefer — APPROVED** [review]

File: pkg/nats/nats.go [hunk 1]
```diff
@@ -0,0 +1,323 @@
...
+
+// CreateTLSConfig creates a TLS Config for NATS connection.
+// implements SecurableConfig from NATS package.
+func (i *InMemorySecureConfig) CreateTLSConfig() (*tls.Config, error) {
+	return &tls.Config{
+		ServerName:           i.serverName,
+		MinVersion:           tls.VersionTLS13,
+		GetClientCertificate: i.certGetter,
+		//nolint:gosec // manually verify this cert
+		InsecureSkipVerify: true,
```
[the remark this answers is not in the document]
Comment (sjberman) [reply pkg/nats/nats.go:179]: Dismissing due to the nolint comment (manual verification)

File: pkg/nats/nats.go [hunk 2]
```diff
@@ -0,0 +1,323 @@
...
+	return nil
+}
+
+// CreateTLSConfig creates a TLS config using the client cert and key
+// in the secure config.
+func (sc OnDiskSecureConfig) CreateTLSConfig() (*tls.Config, error) {
+	tlsConfig := &tls.Config{
+		ServerName: sc.serverName,
+		//nolint:gosec // Skip server certificate verification so that we can manually verify
+		InsecureSkipVerify: true,
```
[the remark this answers is not in the document]
Comment (sjberman) [reply pkg/nats/nats.go:236]: Dismissing due to the nolint comment (manual verification)

**Review by jbyers19 — APPROVED** [review]

File: pkg/nats/nats.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (jbyers19) [reply pkg/nats/nats.go:179]: Would placing the nolint comment on the same line keep this check from being triggered in the future?

[the remark this answers is not in the document]
Comment (sjberman) [reply pkg/nats/nats.go:179]: Not sure. I don't know if CodeQL reads nolint comments. I have a feeling it doesn't.

**Pull request merged**